include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -1162,71 +1162,25 @@ namespace internal
       MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>
         &transfer)
     {
-      const Number *weights = transfer.weights.data();
-
-      // Helper function that extracts the weight. Before doing so,
-      //  it checks the values already set.
-      const auto set = [](auto &mask, const auto &weight) {
-        if (mask == -1.0 || mask == weight)
-          {
-            mask = weight;
-            return true;
-          }
-
-        return false;
-      };
-
-      std::vector<std::array<Number, Utilities::pow(3, dim)>> masks;
-
-      // loop over all schemes
+      unsigned int n_cells = 0;
       for (const auto &scheme : transfer.schemes)
-        {
-          const int    loop_length = scheme.degree_fine + 1;
-          unsigned int degree_to_3[100];
-          degree_to_3[0] = 0;
-          for (int i = 1; i < loop_length - 1; ++i)
-            degree_to_3[i] = 1;
-          degree_to_3[loop_length - 1] = 2;
-
-          // loop over all cells
-          for (unsigned int cell = 0; cell < scheme.n_coarse_cells; ++cell)
-            {
-              std::vector<std::array<Number, Utilities::pow(3, dim)>> mask(
-                transfer.n_components);
-
-              // loop over all components
-              for (unsigned int c = 0; c < transfer.n_components; ++c)
-                mask[c].fill(-1.0);
```

Comment:
Where did this part go? If I read the code correctly, we would now never apply the compression?

Reply:
I have added it to the new function.